### PR TITLE
added base support for using full routes

### LINKED ISSUE #329: URL Generation
It would be useful to provide this functionality : https://laravel.com/docs/5.6/urls
Example:
```
group([
        get('/activate/@token', 'RegisterController@validate').name('validate')
    ], prefix='/auth', name='auth.'),
```
Before
```
Notify.mail(EmailVerificationNotification, to=user.email, link='http://localhost:8000/activate/{0}'.format(token))
```
After
```
Notify.mail(EmailVerificationNotification, to=user.email, link=route('auth.activate', ['token' => token]))
```

### PR DESCRIPTION
**Pull request by josephmancuso**
This adds support for adding "full" urls using the route method on the request class.

A code example would be this:

```python
def show(self, request: Request):
    request.route('dashboard.user') # /dashboard/user
    request.route('dashboard.user', full=True) # http://localhost:8000/dashboard/user
    request.route('dashboard.user', {'id': 1}) # /dashboard/user/1
    request.route('dashboard.user', {'id': 1}, full=True) # http://localhost:8000/dashboard/user/1
    request.route('dashboard.user', {'id': 1}, True) # http://localhost:8000/dashboard/user/1

```

Closes #329

### CODE REVIEW

**Comment by josephmancuso** [comment]
@nioperas06 ☝️

File: masonite/request.py [hunk 1]
```diff
@@ -613,7 +612,8 @@ def route(self, name, params={}):
             masonite.routes.Route|None -- Returns None if the route cannot be found.
```
Comment (clsource) [inline masonite/request.py]: Needs to specify the full=False param and what is it purpose 👍

Comment (clsource) [reply masonite/request.py]: Also adding a convenience function with the full=True would be helpful

```python
def route_full(self, name, params={}):
  return self.route(self, name, params, full=True)
```

Comment (josephmancuso) [reply masonite/request.py]: good catch on the doc string. I'm not too crazy about wrapping the method in another method though

**Pull request merged**